File: src/main/java/fr/vlegall/sochief/service/category/RecipeCategoryMapper.java
package fr.vlegall.sochief.service.category;

import fr.vlegall.sochief.contracts.common.NamedIdDto;
import fr.vlegall.sochief.model.recipe.RecipeCategory;
import org.springframework.stereotype.Component;

@Component
public class RecipeCategoryMapper {
    public NamedIdDto toDto(RecipeCategory recipeCategory) {
        return new NamedIdDto(recipeCategory.getId(), recipeCategory.getName());
        }
}

File: src/main/kotlin/fr/vlegall/sochief/service/category/RecipeCategoryMapper.kt
package fr.vlegall.sochief.service.category

import fr.vlegall.sochief.model.recipe.RecipeCategory
import fr.vlegall.sochief.contracts.common.NamedIdDto
import org.springframework.stereotype.Component

@Component
class RecipeCategoryMapper {
    fun toDto(recipeCategory: RecipeCategory): NamedIdDto =
        NamedIdDto(requireNotNull(recipeCategory.id), requireNotNull(recipeCategory.name))
}